### PR TITLE
[Serve] Add `application` tag to `ray_serve_num_http_error_requests` metric

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

The `Error QPS per Application` tile on the Ray dashboard is empty because the `ray_serve_num_http_error_requests` metric lacks an `"Application"` tag. This change adds the `"Application"` tag to expose the metric on the Ray dashboard.

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This change relies on existing tests.

### CODE REVIEW

**Comment by shrekris-anyscale** [comment]
@alanwguo to verify that the dashboard works with this change.

**Review by GeneDer — APPROVED** [review]
:shipit:

**Comment by alanwguo** [comment]
> @alanwguo to verify that the dashboard works with this change.

verified the fix works.

@shrekris-anyscale , can you also include this small fix in this PR: https://github.com/alanwguo/ray/commit/d697adde2437ca6c3dcbdfc6d2c8ca25dfa713ff

seems like newer versions of prometheus have changed the health check message

**Comment by shrekris-anyscale** [comment]
> can you also include this small fix in this PR

@alanwguo thanks for the find, I added the change.

**Review by edoakes — APPROVED** [review]

**Review by alanwguo — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
All test failures are unrelated. This change is ready to merge.

**Review by richardliaw — APPROVED** [review]

File: dashboard/modules/metrics/metrics_head.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ async def prometheus_health(self, req):
                 text = await resp.text()
                 # Basic sanity check of prometheus health check schema
                 # Different flavors of Prometheus may use different health check strings
-                if "Prometheus" not in text or "OK" not in text:
+                if "Prometheus" not in text:
```
Comment (marijncv) [inline dashboard/modules/metrics/metrics_head.py:168]: Hey @richardliaw, just curious, why was the `"OK"` clause removed in this PR? I'm looking into allowing a VictoriaMetrics backend (see [issue](https://github.com/ray-project/ray/issues/42619)) that would ideally also allow a different string to pass the check.

Maybe even more generally, is the `"Prometheus"` check in the response message really needed? It seems very vulnerable to changes in prometheus, and blocks other flavors/implementations of the Prometheus API (e.g. Thanos & VictoriaMetrics)

Comment (alanwguo) [reply dashboard/modules/metrics/metrics_head.py:168]: Good point. We'll remove this check in the future and only rely on a 200 response.

Comment (alanwguo) [reply dashboard/modules/metrics/metrics_head.py:168]: https://github.com/ray-project/ray/pull/44429

**Pull request merged**